File: app/src/main/java/id/ac/stiki/doleno/absenin/view/admin/ui/my_profile/MyProfileFragment.java
package id.ac.stiki.doleno.absenin.view.admin.ui.my_profile;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.lifecycle.ViewModelProvider;

import dagger.hilt.android.AndroidEntryPoint;
import id.ac.stiki.doleno.absenin.R;
import id.ac.stiki.doleno.absenin.databinding.FragmentMyProfileBinding;
import id.ac.stiki.doleno.absenin.view.dialog.ConfirmDialog;
import id.ac.stiki.doleno.absenin.view.login.LoginActivity;

@AndroidEntryPoint
public class MyProfileFragment extends Fragment {
    private FragmentMyProfileBinding binding;
    private MyProfileViewModel viewModel;
    private ConfirmDialog confirmDialog;

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        binding = FragmentMyProfileBinding.inflate(inflater, container, false);
        confirmDialog = new ConfirmDialog(this.getActivity());

        viewModel = new ViewModelProvider(this).get(MyProfileViewModel.class);
        viewModel.getUser();
        viewModel.myProfileState.observeForever(state -> {
            switch (state) {
                case LOADING:
                    break;
                case SUCCESS:
                case FAILED:
                    binding.etEmail.setText(viewModel.user.email);
                    binding.etName.setText(viewModel.user.name);
                    binding.etRole.setText(viewModel.user.role.getText());
                    break;
            }
        });

        binding.btnLogout.setOnClickListener(v -> {
                    confirmDialog.show(this.getActivity().getString(R.string.logout_message), click -> {
                        Intent intent = new Intent(this.getActivity(), LoginActivity.class);
                        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                        startActivity(intent);
                        viewModel.logout();
                    });
                }
        );

        return binding.getRoot();
    }
}

File: app/src/main/java/id/ac/stiki/doleno/absenin/view/participant/ui/my_profile/MyProfileFragment.kt
package id.ac.stiki.doleno.absenin.view.participant.ui.my_profile

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModelProvider
import dagger.hilt.android.AndroidEntryPoint
import id.ac.stiki.doleno.absenin.R
import id.ac.stiki.doleno.absenin.databinding.FragmentParticipantProfileBinding
import id.ac.stiki.doleno.absenin.view.dialog.ConfirmDialog
import id.ac.stiki.doleno.absenin.view.login.LoginActivity

@AndroidEntryPoint
class MyProfileFragment : Fragment() {
    private lateinit var binding: FragmentParticipantProfileBinding
    private lateinit var viewModel: MyProfileViewModel
    private lateinit var confirmDialog: ConfirmDialog

    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentParticipantProfileBinding.inflate(inflater, container, false)

        confirmDialog = ConfirmDialog(this.activity)

        viewModel = ViewModelProvider(this)[MyProfileViewModel::class.java]
        viewModel.getUser()
        viewModel.myProfileState.observeForever { state ->
            run {
                binding.etEmail.setText(viewModel.user?.email ?: "")
                binding.etName.setText(viewModel.user?.name ?: "")
                binding.etRole.setText(
                    viewModel.user?.role?.text ?: resources.getStringArray(R.array.list_role).last()
                    ?: "Peserta Acara"
                )
            }
        }

        binding.btnLogout.setOnClickListener { v ->
            confirmDialog.show(
                getString(R.string.logout_message)
            ) {
                val intent = Intent(this.activity, LoginActivity::class.java)
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK
                startActivity(intent)
                viewModel.logout()
            }
        }

        return binding.root
    }
}